### PR TITLE
Enable PostgreSQL support for GDAL on EL9

### PR DESCRIPTION
**Pull request by hummeltech**
Fixes error when running `ogr2ogr -f PostgreSQL ...`:
`Unable to find driver 'PostgreSQL'`

Fixes error when building:
`Could NOT find PostgreSQL (missing: PostgreSQL_LIBRARY PostgreSQL_INCLUDE_DIR)`
I.E.: [see example here](https://github.com/radiant-maxar/geoint-deps/actions/runs/3347229207/jobs/5545774788#step:3:2180)

### CODE REVIEW

File: SPECS/el9/gdal.spec [hunk 1]
```diff
@@ -267,6 +267,9 @@ export PYTEST="pytest -vv -p no:sugar --color=no"
 #  * ogr/ogr_pg: trying to use PostGIS when it's not supposed to
 #  * pyscripts/test_gdal2tiles.py: regressions with test file locations on EL.
 $PYTEST \
+--deselect ogr/ogr_fgdb.py::test_ogr_fgdb_19 \
```
Comment (jbronn) [inline SPECS/el9/gdal.spec:270]: I thought issues were fixed with 3.6.0 as they passed fine in a local build.  Then I remembered these specifically can fail randomly, even if they pass intermittently.

**Review by jbronn — APPROVED** [review]
LGTM

**Comment by jbronn** [comment]
Great find on this @hummeltech!  What's funny is that other tests and drivers using PostgreSQL, specifically [`PGDump`](https://gdal.org/drivers/vector/pgdump.html), were enabled.  My hunch is that the new CMake build system of 3.5.0 doesn't autodetect PostgreSQL like it used, hence why you even gotta specify include/library paths despite `pg_config` being available for that; probably a bug [`FindPostgreSQL.cmake`](https://github.com/OSGeo/gdal/blob/master/cmake/modules/3.20/FindPostgreSQL.cmake) that doesn't take into account the way PGDG does things.

**Comment by hummeltech** [comment]
Yeah, I was getting the tile server moved to EL9 and noticed this breaking issue. Awesome!

**Pull request merged**